stratum/lib/security/credentials_manager.h

```diff
@@ -8,95 +9,67 @@
-                             std::string server_private_key,
-                             std::string server_cert);
-
-  // Public methods from TlsCredentialReloadInterface
-  int Schedule(TlsCredentialReloadArg* arg) override
-      LOCKS_EXCLUDED(credential_lock_);
-  void Cancel(TlsCredentialReloadArg* arg) override;
-
-  // Loads new credentials
-  ::util::Status LoadNewCredential(const std::string ca_cert,
-                                   const std::string cert,
-                                   const std::string key)
-      LOCKS_EXCLUDED(credential_lock_);
-
-  // CredentialsReloadInterface is neither copyable nor movable.
-  CredentialsReloadInterface(const CredentialsReloadInterface&) = delete;
-  CredentialsReloadInterface& operator=(const CredentialsReloadInterface&) =
-      delete;
+DECLARE_string(ca_cert_file);
+DECLARE_string(server_key_file);
+DECLARE_string(server_cert_file);
 
- private:
-  absl::Mutex credential_lock_;
-  bool reload_credential_ GUARDED_BY(credential_lock_);
-  std::string pem_root_certs_ GUARDED_BY(credential_lock_);
-  std::string server_private_key_ GUARDED_BY(credential_lock_);
-  std::string server_cert_ GUARDED_BY(credential_lock_);
-};
+namespace stratum {
 
 // CredentialsManager manages the server credentials for (external facing) gRPC
 // servers. It handles starting and shutting down TSI as well as generating the
-// server credentials. This class is supposed to be created
-// once for each binary.
+// server credentials. This class is supposed to be created once for each
+// binary.
 class CredentialsManager {
  public:
   virtual ~CredentialsManager();
 
-  // Generates server credentials for an external facing gRPC
-  // server.
+  // Generates server credentials for an external facing gRPC server.
   virtual std::shared_ptr<::grpc::ServerCredentials>
   GenerateExternalFacingServerCredentials() const;
 
+  // Generates client credentials for contacting an external facing gRPC server.
+  virtual std::shared_ptr<::grpc::ChannelCredentials>
+  GenerateExternalFacingClientCredentials() const;
+
+  // Loads new server credentials.
+  virtual ::util::Status LoadNewServerCredentials(const std::string& ca_cert,
+                                                  const std::string& cert,
+                                                  const std::string& key);
+
+  // Loads new client credentials.
+  virtual ::util::Status LoadNewClientCredentials(const std::string& ca_cert,
+                                                  const std::string& cert,
+                                                  const std::string& key);
+
   // Factory functions for creating the instance of the class.
-  static ::util::StatusOr<std::unique_ptr<CredentialsManager>> CreateInstance();
+  // If TLS certificates are unavailable, the flag secure_only=false allows insecure_mode for gRPC port
+  static ::util::StatusOr<std::unique_ptr<CredentialsManager>> CreateInstance(bool secure_only=false);
```

Comment:
Introduced by me: A `secure_only` flag that defaults to `false` to maintain backward compatibility